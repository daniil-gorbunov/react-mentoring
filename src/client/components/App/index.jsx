import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Grid, Row, Col } from 'react-flexbox-grid';
import SearchPage from '../SearchPage';

export default () => (
  <Router>
    <Grid>
      <Row>
        <Col xs={12}>
          <Route path="/" component={SearchPage} />
        </Col>
      </Row>
    </Grid>
  </Router>
);
