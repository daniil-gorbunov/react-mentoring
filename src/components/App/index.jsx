import React from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import { Grid, Row, Col } from 'react-flexbox-grid';
import SearchPage from '../SearchPage';

const App = () => (
  <Router>
    <Grid>
      <Row>
        <Col xs={12}>
          <ul>
            <li><Link to="/">Home</Link></li>
            <li><Link to="/search/qwe">Search</Link></li>
            <li><Link to="/film/some film">Film</Link></li>
          </ul>
          <Route path="/" component={SearchPage} />
        </Col>
      </Row>
    </Grid>
  </Router>
);

export default App;
