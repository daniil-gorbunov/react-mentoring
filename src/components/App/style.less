html, body {
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  margin: 0;
  padding: 0;
}

.app {
  width: 1200px;
  margin: 0 auto;
  color: #fff;
}

.uppercase {
  text-transform: uppercase;
}

.row {
  margin-top: 25px;
}

button {
  border: 0;
  text-transform: uppercase;
  padding: 5px 30px;
  border-radius: 3px;
  cursor: pointer;
}
