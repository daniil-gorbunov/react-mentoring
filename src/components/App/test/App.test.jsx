import React from 'react';
import renderer from 'react-test-renderer';
import App from '..';

jest.mock('react-router-dom', () => ({
  BrowserRouter: 'BrowserRouter',
  Route: 'Route',
}));

describe('App', () => {
  test('renders correctly', () => {
    const app = renderer.create(<App />);
    expect(app).toMatchSnapshot();
  });
});
