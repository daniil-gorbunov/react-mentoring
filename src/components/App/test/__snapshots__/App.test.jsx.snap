// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<BrowserRouter>
  <Grid>
    <Row>
      <Col
        xs={12}
      >
        <Route
          component={[Function]}
          path="/"
        />
      </Col>
    </Row>
  </Grid>
</BrowserRouter>
`;
